mqtt/mqtt.js

```diff
@@ -80,14 +80,27 @@ function startMqtt() {
     return net.createServer(aedes.handle).listen(config.mqtt_port);
 }
 
-nats.subscribe('channel.*', {'queue':'mqtts'}, function (msg) {
+function parseMqttTopic(topic) {
+    // allow channels/any-char-allowed/message with 0|+ block /any-char-allowed
+    // the text messages and the last subtopic can not terminate with /
+    return /^channels\/(.+?)\/messages(\/.+[^\/]$)*$/.exec(topic)
+}
+
+nats.subscribe('channel.>', {'queue':'mqtts'}, function (msg) {
     var m = message.RawMessage.decode(Buffer.from(msg)),
-        packet;
+        packet, channelParts;
     if (m && m.protocol !== 'mqtt') {
+        // rexexp has two element, the first one is the channelId, wile
+        // the second is undefined or the subtopic parts starting from the dot
+        channelParts = /^([\w\-]+)(\..+[^\.]$)*$/.exec(m.channel)
```

Comment:
@drasko AFAIK earlier we used `channel` only for channel ID. This means that we'll store whole topic in `channel` field, which is not only channel ID. Just want you to be aware of this.

Reply:
Why wouldn't it?

Reply:
Sorry, didn't realized that there was a dot there. But still, I didn't know that you've already decided that you're going to pass subtopic here too. Maybe the rest of the team should be in sync with this change too.

Reply:
Yep, subtopics will be enabled for all 4 adapters. @nmarcetic - remainder to open the issues.

Reply:
That's OK, but there is no spec how will this be done, that's why I don't know if this is how we are going to do it. So we are going to pass `<channel_id>.subtopic` in `channel` field? Why not adding separate field that will be called subtopic, and leave `channel` field as is? Then there will be no need to extract `channel_id` from string.

Reply:
Yes, that would make code probably cleaner. But to see exactly what you propose, can you write here a small snippet example please.

Reply:
So, instead of using current `channel` field from here, I would add a new field: `subtopic`. So our structure would look like this:
```
message RawMessage {
	string channel     = 1;
        string subtopic = 2;
	string publisher   = 3;
	string protocol    = 4;
	string contentType = 5;
	bytes  payload     = 6;
}
```
In `channel`, we would store just channel ID, like we've done so far, and in `subtopic` we would store everything else (subtopic).